articles/internet-protocols/index.md

```diff
@@ -0,0 +1,133 @@
+- A beginner's understanding of computer networking.
+
+- Basic knowledge of internet communication and data transmission.
+
+
+### Introduction to IPv4 and IPv6
+
+IPv4 and IPv6 are the actual protocols tasked with data transmission in the form of packets and datagrams. 
+
+IPv4 protocol is mainly used with [ethernets](https://en.wikipedia.org/wiki/Ethernet)	during packet switching in the link-layer networks.	
+
+IPv6 protocol being more newfangled has improved capabilities as compared to IPV4. It can even have infinite addresses hence is replacing IPV4 due to the daily growth in technology increase number of networks being set up frequently.
+
+### Internet protocol version 4
+
+Internet protocol version 4, just as its name suggests, was the fourth version of the IP suite developed by the [DARPA](https://www.darpa.mil/work-with-us/technology-demonstrations#:~:text=DARPA%20is%20developing%20technologies%20to,deriving%20insights%20from%20diverse%20datasets.&text=The%20electromagnetic%20spectrum%20functions%20as,and%20voice%20of%20modern%20society.) and released for use in 1982. It is a significant protocol of the standard networking protocols on the internet and all the other packet switching networks. 
+#### IP Header
+
+![IPv4 Communication Protocol](Screenshot_20210428-104617.png)
+
+IPv4 still controls the internet traffics despite the development of IPv6 using a 32bit address space providing over four million unique addresses.
+
+IPv4 being a connectionless protocol, does not guarantee if data is delivered, neither does it arrange the data packets properly. Some packets may also have duplicates, so the organization is all addressed by the layer that handles transport, i.e., [TCP](https://www.sdxcentral.com/resources/glossary/transmission-control-protocol-tcp/#:~:text=Transmission%20Control%20Protocol%20(TCP)%20%E2%80%93,referred%20to%20as%20TCP%2FIP.)or the [UDP](https://en.wikipedia.org/wiki/User_Datagram_Protocol). IPv4 also has special and multicast addresses for [private](https://en.wikipedia.org/wiki/Private_network) networks with approximately eighteen million addresses.
+
+
+IPv4 address digits are separated with decimal points as shown below:
+
+#### Examples of IPv4 addresses
+
+               172.16.254.1
+
+               169.254.255.255 
+### IPv4 addressing
+
+Three different types of addressing modes supported in IPv4, namely; 
+- Unicast addressing mode
+- Broadcast addressing mode
+- Multicast addressing mode
+                           
+
+### Internet protocol version 6
+
+Internet protocol version 6 is the newest version of the internet protocol suite as of now developed to replace the fourth version, i.e., the IPv4. It was brought into existence by the [IETF](https://www.ietf.org/blog/ipv6-internet-standard/) because IPv4 has exhausted its addresses. IPv6 was intended to replace IPv4. However, this has not been the case; IPv4 has continued to live on.
```

Comment:
```suggestion
Internet protocol version 6 is the newest version of the internet protocol suite as of now developed to replace the fourth version, i.e., the IPv4. IPv6 was brought into existence by the [IETF](https://www.ietf.org/blog/ipv6-internet-standard/) because IPv4 had exhausted its addresses. IPv6 was intended to replace IPv4. However, this has not been the case; IPv4 has continued to live on.